antlr-kotlin-runtime/src/commonMain/kotlin/org/antlr/v4/kotlinruntime/tree/Trees.kt

```diff
@@ -76,24 +76,19 @@ public object Trees {
 
   public fun getNodeText(t: Tree, ruleNames: List<String>?): String {
     if (ruleNames != null) {
-      if (t is RuleContext) {
-        val ruleIndex = t.ruleContext.ruleIndex
-        val ruleName = ruleNames[ruleIndex]
-        val altNumber = t.altNumber
-
-        return if (altNumber != ATN.INVALID_ALT_NUMBER) {
-          "$ruleName:$altNumber"
-        } else {
-          ruleName
-        }
-      } else if (t is ErrorNode) {
-        return t.toString()
-      } else if (t is TerminalNode) {
-        val symbol = t.symbol
-
-        if (symbol != null) {
-          return symbol.text!!
+      when (t) {
+        is RuleContext -> {
+          val ruleIndex = t.ruleContext.ruleIndex
+          val ruleName = ruleNames[ruleIndex]
+          val altNumber = t.altNumber
+          return if (altNumber != ATN.INVALID_ALT_NUMBER) {
+            "$ruleName:$altNumber"
+          } else {
+            ruleName
+          }
         }
+        is ErrorNode -> return t.toString()
+        is TerminalNode -> return t.symbol.text!!
```

Comment:
Should we write something along the lines of:
```
t.symbol.text ?: throw IllegalStateException("Token symbol text should not be null")
```

Wouldn't it be null for EOF?

Reply:
Yes that's better, I'll commit the change.